drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -890,6 +890,32 @@ static int esp32_wifi_dev_init(const struct device *dev)
 	return 0;
 }
 
+static int esp32_wifi_set_config(const struct device *dev, enum ethernet_config_type type,
+				 const struct ethernet_config *config)
+{
+	struct esp32_wifi_runtime *dev_data = dev->data;
+
+	switch (type) {
+	case ETHERNET_CONFIG_TYPE_MAC_ADDRESS:
+		esp_wifi_set_mode(ESP32_WIFI_MODE_STA);
+		memcpy(dev_data->mac_addr, config->mac_address.addr, 6);
+		net_if_set_link_addr(esp32_wifi_iface, dev_data->mac_addr,
+				     sizeof(dev_data->mac_addr), NET_LINK_ETHERNET);
+
+		esp_err_t ret = esp_wifi_set_mac(WIFI_IF_STA, dev_data->mac_addr);
+
+		if (ret != ESP_OK) {
+			printk("Failed to set mac address: %d\n", ret);
+		}
```

Comment:
I think it would be better if we call
```
		memcpy(dev_data->mac_addr, config->mac_address.addr, 6);
		net_if_set_link_addr(esp32_wifi_iface, dev_data->mac_addr,
				     sizeof(dev_data->mac_addr), NET_LINK_ETHERNET);
```
after the `esp_wifi_set_mac()` has succeeded. Otherwise if the call fails, we have inconsistent mac address in upper net stack compared to esp32 driver.

Reply:
Well, let's update it now.